### DISCUSSION TITLE
What is the impact on the lifecycle of fingerprintjs

### DISCUSSION THREAD

**Discussion by fyjiayou** [fingerprintjs/fingerprintjs#996] [Q&A] [answered]
What is the impact on the lifecycle of fingerprintjs? Can the lifecycle of the device ID be extended?

**Comment by JuroUhlar** [comment, marked as answer]
Hi @fyjiayou! Can you expand on the question, please? I am not sure I understand what you mean by "lifecycle". Are you trying to make the FingerprintJS browser identifier last longer, make it more stable?

It's always a balance between longevity/stability and false positives. [Removing less stable components](https://github.com/fingerprintjs/fingerprintjs/blob/master/docs/extending.md) of the fingerprint will in principle increase the longevity of the identifier (how long it stays the same before some change in the browser resets it) but it will also likely increase false positives (situations where two different browsers are assigned the same identifier, because now less signals can be used to differentiate them).

The right balance is usually different for different use cases. Getting a more stable identifier without increasing false positives is one of the things that [Fingerprint Pro does significantly better than FingerprintJS](https://fingerprint.com/github/).

**Reply by fyjiayou** [reply]
Thank you for answering my question
